src/library-authoring/data/apiHooks.ts

```diff
@@ -210,3 +212,23 @@ export const useUpdateXBlockFields = (contentLibraryId: string, usageKey: string
     },
   });
 };
+
+/**
+ * Use this mutation to create a library collection
+ */
+export const useCreateLibraryCollection = (libraryId: string | undefined) => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: (data: CreateLibraryCollectionDataRequest) => {
+      if (libraryId) {
+        return createCollection(libraryId, data);
+      }
+      return Promise.resolve();
+    },
+    onSettled: () => {
+      if (libraryId) {
+        queryClient.invalidateQueries({ predicate: (query) => libraryQueryPredicate(query, libraryId) });
+      }
+    },
+  });
+};
```

Comment:
Why do we need to allow for an undefined `libraryId` here? This method shouldn't be called if we don't have a library.

Reply:
It's because the `libraryId` is obtained from `useParams` and [can be null](https://github.com/openedx/frontend-app-course-authoring/pull/1259/files#diff-d59f0d40974ef056a55204be49fafb7d8df4599374a8ba26177fe87b2899dd99R20), and this is the way that we found to avoid the `React Hook "hook" is called conditionally`

![image](https://github.com/user-attachments/assets/e8f1d7fb-d814-45fc-9af4-c3b8eb51cef1)

Reply:
Can that check be made back [in the caller](https://github.com/openedx/frontend-app-course-authoring/pull/1259/files#diff-d59f0d40974ef056a55204be49fafb7d8df4599374a8ba26177fe87b2899dd99R20), since it doesn't make sense to even show a Create Collection modal if there's no library?

Reply:
I found this solution

```
const { libraryId } = useParams();
const create = useCreateLibraryCollection(libraryId!);
```

I can't add an `if` because of the same error `React Hook "hook" is called conditionally`

Reply:
You can also do this: https://github.com/openedx/frontend-app-course-authoring/blob/9b61037311591f2eb7cf582cc3910dcf0504d785/src/library-authoring/LibraryAuthoringPage.tsx#L121-L124

Reply:
It works. Thanks!